config/bidderinfo.go

```diff
@@ -177,94 +166,344 @@ type SyncerEndpoint struct {
+		return err
+	}
+	if err := validateCapabilities(info.Capabilities, bidderName); err != nil {
+		return err
 	}
 
-	return &copy
+	return nil
 }
 
-// LoadBidderInfoFromDisk parses all static/bidder-info/{bidder}.yaml files from the file system.
-func LoadBidderInfoFromDisk(path string, adapterConfigs map[string]Adapter, bidders []string) (BidderInfos, error) {
-	reader := infoReaderFromDisk{path}
-	return loadBidderInfo(reader, adapterConfigs, bidders)
+func validateMaintainer(info *MaintainerInfo, bidderName string) error {
+	if info == nil || info.Email == "" {
+		return fmt.Errorf("missing required field: maintainer.email for adapter: %s", bidderName)
+	}
+	return nil
 }
 
-func loadBidderInfo(r infoReader, adapterConfigs map[string]Adapter, bidders []string) (BidderInfos, error) {
-	infos := BidderInfos{}
+func validateCapabilities(info *CapabilitiesInfo, bidderName string) error {
+	if info == nil {
+		return fmt.Errorf("missing required field: capabilities for adapter: %s", bidderName)
+	}
 
-	for _, bidder := range bidders {
-		data, err := r.Read(bidder)
-		if err != nil {
-			return nil, err
+	if info.App == nil && info.Site == nil {
+		return fmt.Errorf("at least one of capabilities.site or capabilities.app must exist for adapter: %s", bidderName)
+	}
+
+	if info.App != nil {
+		if err := validatePlatformInfo(info.App); err != nil {
+			return fmt.Errorf("capabilities.app failed validation: %v for adapter: %s", err, bidderName)
 		}
+	}
 
-		info := BidderInfo{}
-		if err := yaml.Unmarshal(data, &info); err != nil {
-			return nil, fmt.Errorf("error parsing yaml for bidder %s: %v", bidder, err)
+	if info.Site != nil {
+		if err := validatePlatformInfo(info.Site); err != nil {
+			return fmt.Errorf("capabilities.site failed validation: %v, for adapter: %s", err, bidderName)
 		}
+	}
+	return nil
+}
 
-		info.Enabled = isEnabledByConfig(adapterConfigs, bidder)
-		infos[bidder] = info
+func validatePlatformInfo(info *PlatformInfo) error {
+	if len(info.MediaTypes) == 0 {
+		return errors.New("mediaTypes should be an array with at least one string element")
 	}
 
-	return infos, nil
-}
+	for index, mediaType := range info.MediaTypes {
+		if mediaType != "banner" && mediaType != "video" && mediaType != "native" && mediaType != "audio" {
+			return fmt.Errorf("unrecognized media type at index %d: %s", index, mediaType)
+		}
+	}
 
-func isEnabledByConfig(adapterConfigs map[string]Adapter, bidderName string) bool {
-	a, ok := adapterConfigs[strings.ToLower(bidderName)]
-	return ok && !a.Disabled
+	return nil
 }
 
-type infoReader interface {
-	Read(bidder string) ([]byte, error)
+func validateSyncer(bidderInfo BidderInfo) error {
+	if bidderInfo.Syncer == nil {
+		return nil
+	}
+
+	for _, v := range bidderInfo.Syncer.Supports {
+		if !strings.EqualFold(v, "iframe") && !strings.EqualFold(v, "redirect") {
+			return fmt.Errorf("syncer could not be created, invalid supported endpoint: %s", v)
```

Comment:
Could this variable `v` be more descriptive? I understand if you want to leave it the way it is, but I tend to like a little more from variable names

Reply:
Renamed